packages/components/src/toolbar-item/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { useToolbarItem } from 'reakit/Toolbar';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ToolbarContext from '../toolbar-context';
+
+function ToolbarItem( { children, ...props }, ref ) {
+	const accessibleToolbarState = useContext( ToolbarContext );
+	// https://reakit.io/docs/composition/#props-hooks
+	const itemProps = useToolbarItem( accessibleToolbarState, { ...props, ref } );
+
+	if ( typeof children !== 'function' ) {
+		// eslint-disable-next-line no-console
+		console.warn( '`ToolbarItem` is a generic headless component that accepts only function children props' );
```

Comment:
We need to introduce a more robust way of handling developer warnings. It's something for separate PR but this one reminds me about it again :)

Ideally, we port what Reakit uses or React internally.

Reply:
That's on my radar! Will work on that later.

Reply:
Opened a PR #19317

Reply:
Awesome, thanks for doing it. I commented on the PR, it looks great. I just left some thoughts about the integration part, which is something we just need to figure out and pick the best way to move forward for 3rd party developers using this new package in their own projects.